subsys/dfu/dfu_multi_image/src/dfu_multi_image.c

```diff
@@ -247,6 +532,20 @@ int dfu_multi_image_done(bool success)
 		err = writer->close(success);
 	}
 
+#ifdef CONFIG_DFU_MULTI_IMAGE_SAVE_PROGRESS
+
+	if (success) {
+		err = settings_delete(FULL_CBOR_HEADER_SETTING_NAME);
+		if (err != 0) {
+			LOG_ERR("Error deleting stored context from NVM %d", err);
+		}
+		err = settings_delete(FULL_IMAGE_FINISHED_SETTING_NAME);
+		if (err != 0) {
+			LOG_ERR("Error deleting stored context from NVM %d", err);
```

Comment:
these errors should be unique and represent what failed being deleting, otherwise a user is going to see one error message that could mean any number of things